src/renderer/dx/DxFontRenderData.cpp

```diff
@@ -137,36 +137,36 @@ DxFontRenderData::DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwr
         THROW_IF_FAILED(textFormat->SetParagraphAlignment(DWRITE_PARAGRAPH_ALIGNMENT_NEAR));
         THROW_IF_FAILED(textFormat->SetWordWrapping(DWRITE_WORD_WRAPPING_NO_WRAP));
 
-        _textFormatMap.insert({ fontInfo, textFormat });
+        _textFormatMap.emplace(_ToMapKey(weight, style, stretch), textFormat);
         return textFormat;
     }
     else
     {
-        return (*textFormatIt).second;
+        return textFormatIt->second;
     }
 }
 
 [[nodiscard]] Microsoft::WRL::ComPtr<IDWriteFontFace1> DxFontRenderData::FontFaceWithAttribute(DWRITE_FONT_WEIGHT weight,
                                                                                                DWRITE_FONT_STYLE style,
                                                                                                DWRITE_FONT_STRETCH stretch)
 {
-    DxFontInfo fontInfo = _defaultFontInfo;
-    fontInfo.SetWeight(weight);
-    fontInfo.SetStyle(style);
-    fontInfo.SetStretch(stretch);
-
-    const auto fontFaceIt = _fontFaceMap.find(fontInfo);
+    const auto fontFaceIt = _fontFaceMap.find(_ToMapKey(weight, style, stretch));
     if (fontFaceIt == _fontFaceMap.end())
     {
+        DxFontInfo fontInfo = _defaultFontInfo;
+        fontInfo.SetWeight(weight);
+        fontInfo.SetStyle(style);
+        fontInfo.SetStretch(stretch);
+
         std::wstring fontLocaleName = UserLocaleName();
         Microsoft::WRL::ComPtr<IDWriteFontFace1> fontFace = fontInfo.ResolveFontFaceWithFallback(_dwriteFactory.Get(), fontLocaleName);
 
-        _fontFaceMap.insert({ fontInfo, fontFace });
+        _fontFaceMap.emplace(_ToMapKey(weight, style, stretch), fontFace);

```

Comment:
```suggestion
        _fontFaceMap.emplace(_ToMapKey(weight, style, stretch), std::move(fontFace));
```

hmm.. these functions are INCREDIBLY SIMILAR.